src/communication/message-factory.ts

```diff
@@ -154,7 +156,7 @@ export abstract class MessageFactory {
       messageResponseId: message.messageId,
       serialize: serialization,
       content: response instanceof Error ? null : prepare(),
-      error: response instanceof Error ? prepare() : null
+      error: response instanceof Error ? prepareError(response) : null
     });
```

Comment:
In the future if the specific change is unrelated, let's do a separate PR/issue for it. Let me know if i am missing something.